packages/wallet-backend-update/src/rafiki/service.ts

```diff
@@ -0,0 +1,439 @@
+    )
+  }
+
+  private async handleOutgoingPaymentCreated(wh: WebHook) {
+    const rapydWalletId = await this.getRapydWalletIdFromWebHook(wh)
+    const amount = this.getAmountFromWebHook(wh)
+
+    if (!this.validateAmount(amount, wh.type)) {
+      return
+    }
+
+    const holdResult = await this.deps.rapydClient.holdLiquidity({
+      amount: this.amountToNumber(amount),
+      currency: amount.assetCode,
+      ewallet: rapydWalletId
+    })
+
+    if (holdResult.data.status?.status !== 'SUCCESS') {
+      holdResult.data.status?.message &&
+        this.deps.logger.error(holdResult.data.status.message)
+      throw new Error(
+        `Unable to hold liquidity on wallet: ${rapydWalletId} on ${EventType.OutgoingPaymentCreated}`
+      )
+    }
+    await this.deps.rafikiClient.depositLiquidity(wh.id)
+
+    this.deps.logger.info(
+      `Succesfully held ${this.amountToNumber(
+        amount
+      )} in ${rapydWalletId}  on ${EventType.OutgoingPaymentCreated}`
+    )
+  }
+
+  private async handleOutgoingPaymentCompleted(wh: WebHook) {
+    const source_ewallet = await this.getRapydWalletIdFromWebHook(wh)
+    const sendAmount = this.getAmountFromWebHook(wh)
+
+    if (!this.validateAmount(sendAmount, wh.type)) {
+      return
+    }
+
+    const releaseResult = await this.deps.rapydClient.releaseLiquidity({
+      amount: this.amountToNumber(sendAmount),
+      currency: sendAmount.assetCode,
+      ewallet: source_ewallet
+    })
+
+    if (releaseResult.data.status?.status !== 'SUCCESS') {
+      releaseResult.data.status?.message &&
+        this.deps.logger.error(releaseResult.data.status.message)
+      throw new Error(
+        `Unable to release amount ${this.amountToNumber(
+          sendAmount
+        )} from ${source_ewallet}`
+      )
+    }
+
+    const transferResult = await this.deps.rapydClient.transferLiquidity({
+      amount: this.amountToNumber(sendAmount),
+      currency: sendAmount.assetCode,
+      destination_ewallet: this.deps.env.RAPYD_SETTLEMENT_EWALLET,
+      source_ewallet
+    })
+
+    if ((transferResult as any).status?.status !== 'SUCCESS') {
+      ;(transferResult as any).status?.message &&
+        this.deps.logger.error((transferResult as any).status.message)
+      throw new Error(
+        `Unable to transfer from ${source_ewallet} into settlement account ${this.deps.env.RAPYD_SETTLEMENT_EWALLET} on ${EventType.OutgoingPaymentCompleted}`
+      )
+    }
+    await this.deps.rafikiClient.withdrawLiqudity(wh.id)
+
+    await this.deps.transactionService.updateTransaction(
+      { paymentId: wh.data.payment.id },
+      { status: 'COMPLETED', value: sendAmount.value }
+    )
+
+    this.deps.logger.info(
+      `Succesfully transfered ${this.amountToNumber(
+        sendAmount
+      )} from ${source_ewallet} to settlement account on ${
+        EventType.OutgoingPaymentCompleted
+      }`
+    )
+  }
+
+  private async handleOutgoingPaymentFailed(wh: WebHook) {
```

Comment:
I believe it would be a good approach to specify the return type of a function, even if it's `void`. In later versions, the default return type will be `undefined`.

https://devblogs.microsoft.com/typescript/announcing-typescript-5-1-rc/#easier-implicit-returns-for-undefined-returning-functions

Reply:
Will keep in mind.
Also 
@rico191013  @dragosp1011